src/components/ContextMenuItem.js

```diff
@@ -47,62 +46,53 @@ const defaultProps = {
-        // We may want to replace this check by checking the Result from OnPress Callback in future.
-        if (this.props.successIcon || this.props.successText) {
-            this.props.toggleDelayButtonState(this.props.autoReset);
+        if (successIcon || successText) {
+            toggleDelayButtonState(autoReset);
         }
-    }
-
-    render() {
-        const icon = this.props.isDelayButtonStateComplete ? this.props.successIcon || this.props.icon : this.props.icon;
-        const text = this.props.isDelayButtonStateComplete ? this.props.successText || this.props.text : this.props.text;
-        return this.props.isMini ? (
-            <BaseMiniContextMenuItem
-                tooltipText={text}
-                onPress={this.triggerPressAndUpdateSuccess}
-                isDelayButtonStateComplete={this.props.isDelayButtonStateComplete}
-            >
-                {({hovered, pressed}) => (
-                    <Icon
-                        small
-                        src={icon}
-                        fill={StyleUtils.getIconFillColor(getButtonState(hovered, pressed, this.props.isDelayButtonStateComplete))}
-                    />
-                )}
-            </BaseMiniContextMenuItem>
-        ) : (
-            <MenuItem
-                title={text}
-                icon={icon}
-                onPress={this.triggerPressAndUpdateSuccess}
-                wrapperStyle={styles.pr9}
-                success={this.props.isDelayButtonStateComplete}
-                description={this.props.description}
-                descriptionTextStyle={styles.breakAll}
-                style={getContextMenuItemStyles(this.props.windowWidth)}
-            />
-        );
-    }
+    }, [isDelayButtonStateComplete, onPress, successIcon, successText, autoReset, toggleDelayButtonState]);
+
+    const icon = isDelayButtonStateComplete ? successIcon || props.icon : props.icon;
+    const text = isDelayButtonStateComplete ? successText || props.text : props.text;
+
+    return props.isMini ? (
+        <BaseMiniContextMenuItem
+            tooltipText={text}
+            onPress={triggerPressAndUpdateSuccess}
+            isDelayButtonStateComplete={isDelayButtonStateComplete}
+        >
+            {({hovered, pressed}) => (
+                <Icon
+                    small
+                    src={icon}
+                    fill={StyleUtils.getIconFillColor(getButtonState(hovered, pressed, isDelayButtonStateComplete))}
+                />
+            )}
+        </BaseMiniContextMenuItem>
+    ) : (
+        <MenuItem
+            title={text}
+            icon={icon}
+            onPress={triggerPressAndUpdateSuccess}
+            wrapperStyle={styles.pr9}
+            success={isDelayButtonStateComplete}
+            description={props.description}
+            descriptionTextStyle={styles.breakAll}
+            style={getContextMenuItemStyles(windowWidth)}
+        />
+    );
```

Comment:
1. JSX could be easier
2. Use `description` instead of `props. description`
```suggestion
    if (props.isMini) {
        return (
        <BaseMiniContextMenuItem
            tooltipText={text}
            onPress={triggerPressAndUpdateSuccess}
            isDelayButtonStateComplete={isDelayButtonStateComplete}
        >
            {({hovered, pressed}) => (
                <Icon
                    small
                    src={icon}
                    fill={StyleUtils.getIconFillColor(getButtonState(hovered, pressed, isDelayButtonStateComplete))}
                />
            )}
        </BaseMiniContextMenuItem>
    )
    }

    return (
        <MenuItem
            title={text}
            icon={icon}
            onPress={triggerPressAndUpdateSuccess}
            wrapperStyle={styles.pr9}
            success={isDelayButtonStateComplete}
            description={description}
            descriptionTextStyle={styles.breakAll}
            style={getContextMenuItemStyles(windowWidth)}
        />
    );
```

Reply:
> 1. JSX could be easier
> 2. Use `description` instead of `props. description`

There is a preference for inline ternaries, probably you could create a thread about this

Reply:
Well, then just: Use `description` instead of `props.description`. Feel free to resolve these conversations :)